src/details/ArborX_DetailsBruteForceImpl.hpp

```diff
@@ -136,8 +142,80 @@ struct BruteForceImpl
               });
         });
   }
+
+  template <class ExecutionSpace, class Predicates, class Values,
+            class Indexables, class Callback>
+  static void query(NearestPredicateTag, ExecutionSpace const &space,
+                    Predicates const &predicates, Values const &values,
+                    Indexables const &indexables, Callback const &callback)
+  {
+    Kokkos::Profiling::ScopedRegion guard("ArborX::BruteForce::query::nearest");
+
+    using MemorySpace = typename Values::memory_space;
+
+    int const n_indexables = values.size();
+    int const n_predicates = predicates.size();
+
+    NearestBufferProvider<MemorySpace> buffer_provider(space, predicates);
+
+    Kokkos::parallel_for(
+        "ArborX::BruteForce::query::nearest::"
+        "check_all_predicates_against_all_indexables",
+        Kokkos::RangePolicy<ExecutionSpace>(space, 0, n_predicates),
+        KOKKOS_LAMBDA(int i) {
+          auto const &predicate = predicates(i);
+          auto const k = getK(predicate);
+          auto const buffer = buffer_provider(i);
+
+          if (k < 1)
+            return;
+
+          using PairIndexDistance =
+              typename NearestBufferProvider<MemorySpace>::PairIndexDistance;
```

Comment:
As a follow up we should think of making this a struct with named parameters.
This is really ugly below when we refer to the "second" to signify the distance.

Reply:
That would be fine with me. The thing I am starting to dislike is having all those PairValueIndex, PairIndexRank, PairIndexDistance thingies floating around. Wonder if there's a better way to handle that.